code/modules/projectiles/ammunition/magazines.dm

```diff
@@ -511,4 +511,32 @@
 	ammo_type = /obj/item/ammo_casing/cap
 	matter = list(DEFAULT_WALL_MATERIAL = 600)
 	max_ammo = 7
-	multiple_sprites = 1
\ No newline at end of file
+	multiple_sprites = 1
+
+
+/obj/item/ammo_magazine/clip/c12g
+	name = "ammo clip (12g slug)"
+	icon_state = "12gclip_s" //largely a codersprite, looks good enough. feel free to make a better one.
+	desc = "A color-coded metal clip for holding and quickly loading shotgun shells. This one is loaded with slugs."
+	caliber = "12g"
+	ammo_type = /obj/item/ammo_casing/a12g
+	matter = list(DEFAULT_WALL_MATERIAL = 1790) // slugs shells x4 + 350 metal for the clip itself.
+	max_ammo = 4
+
+/obj/item/ammo_magazine/clip/c12g/pellet
+	name = "ammo clip (12g buckshot)"
+	icon_state = "12gclip_p"
+	desc = "A color-coded metal clip for holding and quickly loading shotgun shells. This one is loaded with buckshot."
+	ammo_type = /obj/item/ammo_casing/a12g/pellet
+	matter = list(DEFAULT_WALL_MATERIAL = 1790) // buckshot and slugs cost the same
+
+/obj/item/ammo_magazine/clip/c12g/beanbag
+	name = "ammo clip (12g beanbag)"
+	icon_state = "12gclip_bean"
+	desc = "A color-coded metal clip for holding and quickly loading shotgun shells. This one is loaded with beanbags."
+	ammo_type = /obj/item/ammo_casing/a12g/beanbag
+	matter = list(DEFAULT_WALL_MATERIAL = 1070) //beanbags x5 + 350 metal
```

Comment:
yep. 

some day in the dark, dark future, my typo will confuse a new coder.

and they shall suffer the same as i.